include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -187,12 +187,13 @@ class argument_parser
         subcommands{std::move(subcommands)}
     {
         if (!std::regex_match(app_name, app_name_regex))
-            throw design_error{"The application name must only contain alpha-numeric characters or '_' and '-' "
-                               "(regex: \"^[a-zA-Z0-9_-]+$\")."};
+            throw design_error{("The application name must only contain alpha-numeric characters or '_' and '-' "
+                               "(regex: \"^[a-zA-Z0-9_-]+$\").")};
 
         for (auto & sub : this->subcommands)
-            if (!std::regex_match(sub, std::regex{"^[a-zA-Z0-9_]+$"}))
-                throw design_error{"The subcommand name must only contain alpha-numeric characters or '_'."};
+            if (!std::regex_match(sub, app_name_regex))
+                throw design_error{("The subcommand name must only contain alpha-numeric characters or '_' and '-' "
+                                   "(regex: \"^[a-zA-Z0-9_-]+$\").")};
```

Comment:
```suggestion
            if (!std::regex_match(sub, app_name_regex))
            {
                throw design_error{"The subcommand name must only contain alpha-numeric characters or '_' and '-' "
                                   "(regex: \"^[a-zA-Z0-9_-]+$\")."};
             }
```

we don't need the parentheses for the string, but the braces for `if` and `for`